src/sparseml/pytorch/optim/modifier.py

```diff
@@ -600,12 +689,25 @@ def scheduled_log_update(
         if not self._enabled:
             raise RuntimeError("modifier must be enabled")
 
+        scheduled_log = (self._log_frequency is not None) and (
```

Comment:
self._log_frequency = -1.0 means logging in every step/epoch right or not logging at all?